addons/account/models/account_account.py

```diff
@@ -253,9 +275,13 @@ def _check_journal_consistency(self):
                 account=account.display_name
             ))
 
-    @api.constrains('company_id')
+    @api.constrains('company_ids')
     def _check_company_consistency(self):
-        for company, accounts in tools.groupby(self, lambda account: account.company_id):
+        accounts_by_company = defaultdict(lambda: self.env['account.account'])
+        for account in self:
+            for company in account.company_ids:
+                accounts_by_company[company] |= account
+        for company, accounts in accounts_by_company.items():
             if self.env['account.move.line'].search_count([
                 ('account_id', 'in', [account.id for account in accounts]),
                 '!', ('company_id', 'child_of', company.id)
```

Comment:
This might fail if you don't have access to all the companies of the account

Reply:
That's right - I'll sudo for the win!